pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java

```diff
@@ -352,6 +336,59 @@ public List<SegmentConversionResult> executeTask(PinotTaskConfig pinotTaskConfig
     }
   }
 
+  private int parallelDownloadAndUntarSegments(String[] downloadURLs, String[] segmentNames, String tableNameWithType,
+      String taskType, File tempDataDir, List<File> inputSegmentDirs)
+      throws Exception {
+    AtomicInteger numRecords = new AtomicInteger(0);
+    List<Future<Void>> futures = new ArrayList<>();
+
+    int nThreads = Math.min(downloadURLs.length, Runtime.getRuntime().availableProcessors());
```

Comment:
I think instead of Runtime.getRuntime().availableProcessors() we should use _minionConf here. 

because there will be many parallel tasks running together. dependent on numConcurrentTasksPerInstance. so if each task uses N threads, it can lead to starvation. Also each task type will have its own numConcurrentTasksPerInstance defined . 

So I think safer to use a minion conf.

Reply:
actually for OSS, we should avoid changing current behaviour by default. So by default we shouldn't create a threadpool but serially process it (same as today). 

If config is present, we use the threadpool with configured number of threads. 

Also I think better to use a common thread pool which is part of the BaseMultipleSegmentsConversionExecutor class. So that every call to executeTask() doesn't re-create the thread pool. We create this thread pool only if config is set.